lib/rubocop/cop/style/documentation.rb

```diff
@@ -47,18 +47,25 @@ def check(ast, ast_with_comments)
           end
         end
 
+        def module_definition?(node)
+          # 'Const = Class.new' or 'Const = Module.new' are module definitions
+          node.class_type? ||
+            node.module_type? ||
+            (node.casgn_type? &&
+              (rhs = node.children[2]).send_type? &&
```

Comment:
I'm trying to come up with a way to make complex matching operations more concise. I've created a `Pattern` class which does recursive pattern matching on AST nodes. Will send a PR soon so you can have a look.